msgpack-core/src/main/java/org/msgpack/core/MessageUnpacker.java

```diff
@@ -1005,9 +1004,10 @@ public String unpackString() throws IOException {
                         sb.append(decodeBuffer);
 
                         decodeBuffer.clear();
-                        cursor += readLen;
-                        consume(readLen);
                     }
+
+                    cursor += readLen;
```

Comment:
Ah. I was wrong. `readLen` can be retrieved by `bb.position()` since we created a new ByteBuffer at 
```
ByteBuffer bb = buffer.toByteBuffer(position, readLen);
```
whose initial position is 0.